mycpp/gc_linked_list.h

```diff
@@ -0,0 +1,170 @@
+#ifndef MYCPP_GC_LINKED_LIST_H
+#define MYCPP_GC_LINKED_LIST_H
+
+#include "mycpp/common.h"  // DCHECK
+#include "mycpp/gc_obj.h"  // ObjHeader, maskbit
+
+template <typename T>
+class LinkedListNode {
+ public:
+  LinkedListNode() = delete;  // A list node with no object makes no sense.
+  LinkedListNode(T obj) : obj_(obj), prev_(nullptr), next_(nullptr) {
+  }
+
+  T obj() const {
+    return obj_;
+  }
+
+  LinkedListNode* prev() const {
+    return prev_;
+  }
+  LinkedListNode* next() const {
+    return next_;
+  }
+
+  // Replace the nodes predecessor with the given node.
+  void set_prev(LinkedListNode* node) {
+    prev_ = node;
+  }
+
+  // Replace the nodes sucessor with the given node.
+  void set_next(LinkedListNode* node) {
+    next_ = node;
+  }
+
+  static constexpr ObjHeader obj_header() {
+    return ObjHeader::ClassFixed(field_mask(), sizeof(LinkedListNode<T>));
+  }
+
+  static constexpr uint32_t field_mask() {
+    return maskbit(offsetof(LinkedListNode, obj_)) |
```

Comment:
This is tricky because whether this bit is set depends on whether T is a pointer or not

For Slab<T> and Tuple{2,3,4} I think we're using `std::is_pointer<T>` at compile time to conditionally set the mask!  That took a bit of figuring out ...

Reply:
Nice catch